### PR TITLE
Updated docs with proposed design.

### PR DESCRIPTION
**Pull request by seizadi**
In the case of authSource for db-controller I document existing design.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -227,7 +318,10 @@ DatabaseClaim:
       - [Host]: The optional host name where the database instance is located.  If the value is omitted, then the host value from the matching InstanceLabel will be used.
       - [Port]: The optional port to use for connecting to the host.  If the value is omitted, then the host value from the matching InstanceLabel will be used.
       - DatabaseName: The name of the database instance.
-
+      - Shape: The optional Shape values are arbitrary and help drive instance selection
+      - CPUs: The optional CPUs value requests the database host cpu core capacity
+      - RAM: The optional RAM value requests the database host memory capacity
```
Comment (daniel-garcia) [inline docs/index.md:323]: Lets not expose RAM or CPU options to end users. They pick a shape.. nothing else.

File: docs/index.md [hunk 2]
```diff
@@ -227,7 +318,10 @@ DatabaseClaim:
...
+      - Storage: The optional Storage value requests the database host storage capacity
```
Comment (daniel-garcia) [inline docs/index.md:324]: This should be min-storage. The controller may over provision storage and thats ok.

File: docs/index.md [hunk 3]
```diff
@@ -156,6 +245,8 @@ data:
       passwordSecretRef=athena-hostapp-password
 ```
 
+* authSource: Determines how database host master password is retrieved. The possible values are "secret" and "aws". In the case of "secret" the value from the passwordSecretRef Secret is used for the password. In the case of "aws" the RDS password is retrieved using AWS APIs and db-controller should have IAM credentials to make the necessary calls.
```
Comment (daniel-garcia) [inline docs/index.md:248]: this is something the controller should decide based on its configuration

**Review by daniel-garcia — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 3 above)
Comment (seizadi) [reply docs/index.md:248]: I am just documenting current implementation here. The new design will not use this pattern, as it is part of infrastructure concern.

File: docs/index.md (hunk 2 above)
Comment (seizadi) [reply docs/index.md:324]: OK

File: docs/index.md (hunk 1 above)
Comment (seizadi) [reply docs/index.md:323]: OK

**Review by daniel-garcia — APPROVED** [review]

**Review by daniel-garcia — APPROVED** [review]

**Pull request merged**